Investigate swiftshader failures on macos debug

On macos debug there are several validation errors when run under swiftshader. Investigate, fix and enable integration tests on macos debug if possible.

e.g.

```
Testing ./tests/cases/texture_base_mip_level.amber
[ERROR] validation layer (Validation):
 [ VUID-vkFreeMemory-memory-00677 ] Object: 0x2d000000002d (Type = 8) | Cannot call vkFreeMemory on VkDeviceMemory 0x2d000000002d[] that is currently in use by a command buffer. The Vulkan spec states: All submitted commands that refer to memory (via images or buffers) must have completed execution (https://www.khronos.org/registry/vulkan/specs/1.1-extensions/html/vkspec.html#VUID-vkFreeMemory-memory-00677)
[ERROR] validation layer (Validation):
 [ VUID-vkDestroyBuffer-buffer-00922 ] Object: 0x2c000000002c (Type = 9) | Cannot free VkBuffer 0x2c000000002c[] that is in use by a command buffer. The Vulkan spec states: All submitted commands that refer to buffer, either directly or via a VkBufferView, must have completed execution (https://www.khronos.org/registry/vulkan/specs/1.1-extensions/html/vkspec.html#VUID-vkDestroyBuffer-buffer-00922)
[ERROR] validation layer (Validation):
 [ VUID-vkDestroyRenderPass-renderPass-00873 ] Object: 0x1a000000001a (Type = 18) | Cannot call vkDestroyRenderPass on VkRenderPass 0x1a000000001a[] that is currently in use by a command buffer. The Vulkan spec states: All submitted commands that refer to renderPass must have completed execution (https://www.khronos.org/registry/vulkan/specs/1.1-extensions/html/vkspec.html#VUID-vkDestroyRenderPass-renderPass-00873)
[ERROR] validation layer (Validation):
 [ VUID-vkDestroyFramebuffer-framebuffer-00892 ] Object: 0x200000000020 (Type = 24) | Cannot call vkDestroyFramebuffer on VkFramebuffer 0x200000000020[] that is currently in use by a command buffer. The Vulkan spec states: All submitted commands that refer to framebuffer must have completed execution (https://www.khronos.org/registry/vulkan/specs/1.1-extensions/html/vkspec.html#VUID-vkDestroyFramebuffer-framebuffer-00892)
[ERROR] validation layer (Validation):
 [ VUID-vkDestroyImageView-imageView-01026 ] Object: 0x1d000000001d (Type = 14) | Cannot call vkDestroyImageView on VkImageView 0x1d000000001d[] that is currently in use by a command buffer. The Vulkan spec states: All submitted commands that refer to imageView must have completed execution (https://www.khronos.org/registry/vulkan/specs/1.1-extensions/html/vkspec.html#VUID-vkDestroyImageView-imageView-01026)
[ERROR] validation layer (Validation):
 [ VUID-vkDestroyImage-image-01000 ] Object: 0x1b000000001b (Type = 10) | Cannot call vkDestroyImage on VkImage 0x1b000000001b[] that is currently in use by a command buffer. The Vulkan spec states: All submitted commands that refer to image, either directly or via a VkImageView, must have completed execution (https://www.khronos.org/registry/vulkan/specs/1.1-extensions/html/vkspec.html#VUID-vkDestroyImage-image-01000)
[ERROR] validation layer (Validation):
 [ VUID-vkFreeMemory-memory-00677 ] Object: 0x1c000000001c (Type = 8) | Cannot call vkFreeMemory on VkDeviceMemory 0x1c000000001c[] that is currently in use by a command buffer. The Vulkan spec states: All submitted commands that refer to memory (via images or buffers) must have completed execution (https://www.khronos.org/registry/vulkan/specs/1.1-extensions/html/vkspec.html#VUID-vkFreeMemory-memory-00677)
[ERROR] validation layer (Validation):
 [ VUID-vkFreeMemory-memory-00677 ] Object: 0x1f000000001f (Type = 8) | Cannot call vkFreeMemory on VkDeviceMemory 0x1f000000001f[] that is currently in use by a command buffer. The Vulkan spec states: All submitted commands that refer to memory (via images or buffers) must have completed execution (https://www.khronos.org/registry/vulkan/specs/1.1-extensions/html/vkspec.html#VUID-vkFreeMemory-memory-00677)
[ERROR] validation layer (Validation):
 [ VUID-vkDestroyBuffer-buffer-00922 ] Object: 0x1e000000001e (Type = 9) | Cannot free VkBuffer 0x1e000000001e[] that is in use by a command buffer. The Vulkan spec states: All submitted commands that refer to buffer, either directly or via a VkBufferView, must have completed execution (https://www.khronos.org/registry/vulkan/specs/1.1-extensions/html/vkspec.html#VUID-vkDestroyBuffer-buffer-00922)
[ERROR] validation layer (Validation):
 [ VUID-vkDestroyFence-fence-01120 ] Object: 0x190000000019 (Type = 7) | VkFence 0x190000000019[] is in use. The Vulkan spec states: All queue submission commands that refer to fence must have completed execution (https://www.khronos.org/registry/vulkan/specs/1.1-extensions/html/vkspec.html#VUID-vkDestroyFence-fence-01120)
[ERROR] validation layer (Validation):
 [ VUID-vkFreeCommandBuffers-pCommandBuffers-00047 ] Object: 0x7f8cb97b5e40 (Type = 6) | Attempt to free VkCommandBuffer 0x7f8cb97b5e40[] which is in use. The Vulkan spec states: All elements of pCommandBuffers must not be in the pending state (https://www.khronos.org/registry/vulkan/specs/1.1-extensions/html/vkspec.html#VUID-vkFreeCommandBuffers-pCommandBuffers-00047)
[ERROR] validation layer (Validation):
 [ VUID-vkDestroyDescriptorPool-descriptorPool-00303 ] Object: 0x280000000028 (Type = 22) | Cannot call vkDestroyDescriptorPool on VkDescriptorPool 0x280000000028[] that is currently in use by a command buffer. The Vulkan spec states: All submitted commands that refer to descriptorPool (via any allocated descriptor sets) must have completed execution (https://www.khronos.org/registry/vulkan/specs/1.1-extensions/html/vkspec.html#VUID-vkDestroyDescriptorPool-descriptorPool-00303)
[ERROR] validation layer (Validation):
 [ VUID-vkDestroySampler-sampler-01082 ] Object: 0x260000000026 (Type = 21) | Cannot call vkDestroySampler on VkSampler 0x260000000026[] that is currently in use by a command buffer. The Vulkan spec states: All submitted commands that refer to sampler must have completed execution (https://www.khronos.org/registry/vulkan/specs/1.1-extensions/html/vkspec.html#VUID-vkDestroySampler-sampler-01082)
[ERROR] validation layer (Validation):
 [ VUID-vkDestroyImageView-imageView-01026 ] Object: 0x230000000023 (Type = 14) | Cannot call vkDestroyImageView on VkImageView 0x230000000023[] that is currently in use by a command buffer. The Vulkan spec states: All submitted commands that refer to imageView must have completed execution (https://www.khronos.org/registry/vulkan/specs/1.1-extensions/html/vkspec.html#VUID-vkDestroyImageView-imageView-01026)
[ERROR] validation layer (Validation):
 [ VUID-vkDestroyImage-image-01000 ] Object: 0x210000000021 (Type = 10) | Cannot call vkDestroyImage on VkImage 0x210000000021[] that is currently in use by a command buffer. The Vulkan spec states: All submitted commands that refer to image, either directly or via a VkImageView, must have completed execution (https://www.khronos.org/registry/vulkan/specs/1.1-extensions/html/vkspec.html#VUID-vkDestroyImage-image-01000)
[ERROR] validation layer (Validation):
 [ VUID-vkFreeMemory-memory-00677 ] Object: 0x220000000022 (Type = 8) | Cannot call vkFreeMemory on VkDeviceMemory 0x220000000022[] that is currently in use by a command buffer. The Vulkan spec states: All submitted commands that refer to memory (via images or buffers) must have completed execution (https://www.khronos.org/registry/vulkan/specs/1.1-extensions/html/vkspec.html#VUID-vkFreeMemory-memory-00677)
./tests/cases/texture_base_mip_level.amber: Vulkan::Calling vkWaitForFences Timeout
[ERROR] validation layer (Validation):
 [ VUID-vkDestroyDevice-device-00378 ] Object: 0x2a000000002a (Type = 17) | OBJ ERROR : For VkDevice 0x7f8cb97a1008[], VkPipelineLayout 0x2a000000002a[] has not been destroyed. The Vulkan spec states: All child objects created on device must have been destroyed prior to destroying device (https://www.khronos.org/registry/vulkan/specs/1.1-extensions/html/vkspec.html#VUID-vkDestroyDevice-device-00378)
[ERROR] validation layer (Validation):
 [ VUID-vkDestroyDevice-device-00378 ] Object: 0x2b000000002b (Type = 19) | OBJ ERROR : For VkDevice 0x7f8cb97a1008[], VkPipeline 0x2b000000002b[] has not been destroyed. The Vulkan spec states: All child objects created on device must have been destroyed prior to destroying device (https://www.khronos.org/registry/vulkan/specs/1.1-extensions/html/vkspec.html#VUID-vkDestroyDevice-device-00378)

Command '['/tmpfs/src/github/amber/build/amber', '-q', './tests/cases/texture_base_mip_level.amber']' returned non-zero exit status -11
```

--- a/kokoro/macos-clang-debug/build.sh
+++ b/kokoro/macos-clang-debug/build.sh
@@ -17,4 +17,4 @@ set -e  # fail on error
 set -x  # display commands
 
 SCRIPT_DIR=`dirname "$BASH_SOURCE"`
-source $SCRIPT_DIR/../scripts/macos/build.sh DEBUG clang
+source $SCRIPT_DIR/../scripts/macos/build.sh DEBUG clang -DAMBER_ENABLE_SWIFTSHADER=TRUE

